dotnet/src/Connectors/Connectors.AzureAIInference/Core/ChatClientCore.cs

```diff
@@ -463,4 +547,16 @@ private ChatMessageContent GetChatMessage(ChatChoice chatChoice, ChatCompletions
             { nameof(chatChoice.Index), chatChoice.Index },
         };
     }
+
+    private static Dictionary<string, object?> GetResponseMetadata(StreamingChatCompletionsUpdate completions)
+    {
+        return new Dictionary<string, object?>(4)
+        {
+            { nameof(completions.Id), completions.Id },
+            { nameof(completions.Created), completions.Created },
+
+            // Serialization of this struct behaves as an empty object {}, need to cast to string to avoid it.
+            { nameof(completions.FinishReason), completions.FinishReason?.ToString() },
+        };
```

Comment:
```suggestion
        return new Dictionary<string, object?>(3)
        {
            { nameof(completions.Id), completions.Id },
            { nameof(completions.Created), completions.Created },
            // Serialization of this struct behaves as an empty object {}, need to cast to string to avoid it.
            { nameof(completions.FinishReason), completions.FinishReason?.ToString() },
        };
```